src/MultiMessageSender.sol

```diff
@@ -132,18 +142,17 @@ contract MultiMessageSender {
         /// refund remaining fee
         /// FIXME: add an explicit refund address config
         if (address(this).balance > 0) {
-            _safeTransferETH(msg.sender, address(this).balance);
+            _safeTransferETH(gac.getRefundAddress(), address(this).balance);
         }
 
-        emit MultiMessageMsgSent(msgId, nonce, _dstChainId, _target, _callData, _expiration, adapters);
+        emit MultiMessageMsgSent(msgId, nonce, _dstChainId, _target, _callData, msgExpiration, adapters);
```

Comment:
nit: ideally the list of `adapters` in the `MultiMessageMsgSent` event should only include those that didn't fail

Reply:
isn't there a try-catch on line 128?